### ISSUE TITLE
async_alert Doesn't Support Color Reset Ansi Sequences on Windows

### ISSUE THREAD

**Issue by Chris3606** [python-cmd2/cmd2#1313]
# Steps to Reproduce
Color reset sequences (reset foreground, reset background), etc. do not work on Windows when using `async_alert`.

Given the following (baseline) code:

```py 
from cmd2 import Cmd, Fg, style


class App(Cmd):
    def do_print(self, _):
        self.poutput(style("Hello ", fg=Fg.BLUE) + style("World"))

app = App()
app.cmdloop()
```

We can generate sensible colored output:
![image](https://github.com/python-cmd2/cmd2/assets/6090062/d1448155-14d9-454b-b2be-6d8d82f51b59)

However, if we take the [async printing example](https://github.com/python-cmd2/cmd2/blob/master/examples/async_printing.py) and change the first item in the `ALERTS` array to: `style("Hello ", fg=Fg.BLUE) + style("World")` (the same text we used in the original example, it fails rather spectacularly:

![image](https://github.com/python-cmd2/cmd2/assets/6090062/9edaf814-f735-4cef-ab8e-0ab043b99cd3)

# Notes
I believe this is due to a limitation and/or bug in `pyreadline3`.  Its [printing function](https://github.com/pyreadline/pyreadline/blob/4e4b7b468cfdf2dfef8e2c8c695a64e96c514213/pyreadline/console/ansi.py#L70) does not support the color escape sequences at all (only 0, the global reset).

Adding support for the escape sequences is fairly trivial (I plan to ensure a PR is submitted to pyreadline for it).

Given that cmd2 is [already extending pyreadline3](https://github.com/python-cmd2/cmd2/blob/be1593929f046d44e0d27fcd902fdb9998ba6bb3/cmd2/rl_utils.py#L111), however, I wanted to submit for consideration that it may be possible to support this via additions to `rl_utils`.  I plan to investigate this further.

**Comment by Chris3606** [comment]
Actually, this is slightly more extensive than just not having color reset sequences. According to the tests I've ran, `pyreadline3` does not properly support background colors (at all), doesn't support background or foreground reset sequences, and doesn't support underline reset sequences either.

I have put in [an issue and corresponding PR to fix it](https://github.com/pyreadline3/pyreadline3/issues/20) in the pyreadline3 repository, however I'm not sure if it's still maintained or not.

I did test and in my own application was able to "monkey patch" pyreadline3 by creating a class that inherits from `AnsiWriter`, fixing the function, then creating an instance of that class to the `ansiwriter` property of `readline.rl.console.ansiwriter`; though obviously such a patch might have side effects that you don't want if, for instance, a user is using cmd2 and pyreadline3 directly (though this is unlikely).

If you're interested, I'd be happy to submit a PR to patch it, but it seems a bit of a hacky solution.

**Comment by kmvanbrunt** [comment]
I would prefer we wait and see if `pyreadline3` accepts your PR instead of patching their code in `cmd2`.

**Comment by kmvanbrunt** [comment]
As a temporary workaround, does appending `cmd2.TextStyle.RESET_ALL` to your styled text fix the issue?

**Comment by Chris3606** [comment]
Yes, this works at least for the use case above.  Obviously it will reset _all_ elements, so it might not work for nested styles or some such, but I currently have none of those in my code.

**Comment by kmvanbrunt** [comment]
It looks like `pyreadline3` accepted your PR and it's now fixed as of `pyreadline3` version 3.4.2.
I just tested using 3.4.3 in both Windows Command Prompt and Windows Powershell and the `async_alert()` color issue you experienced is working correctly.

Nice job fixing this.
